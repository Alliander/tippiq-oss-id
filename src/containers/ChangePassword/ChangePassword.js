/**
 * Change password container.
 * @module components/ChangePassword
 */

import React, { Component, PropTypes } from 'react';
import Helmet from 'react-helmet';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Grid, Row, Col } from 'react-bootstrap';

import { ChangePasswordForm, PageTitle } from '../../components';
import { changePassword } from '../../actions';

/**
 * Change password container class.
 * @class ChangePassword
 * @extends Component
 */
export class ChangePassword extends Component {

  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    changePassword: PropTypes.func.isRequired,
    password: PropTypes.object.isRequired,
  }

  state = {
    passwordSaved: false,
  };

  /**
   * Component will receive props
   * @method componentWillReceiveProps
   * @param {Object} nextProps Next properties
   * @returns {undefined}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.password.save && nextProps.password.save.success) {
      this.setState({ passwordSaved: true });
    }
  }

  /**
   * Handler for submit action.
   * @method handleSubmit
   * @param {Object} data Form data.
   * @returns {undefined}
   */
  handleSubmit = (data) => {
    this.setState({ passwordSaved: false }, () => this.props.changePassword(data.password));
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    const { password } = this.props;
    const { passwordSaved } = this.state;

    return (
      <div id="page-change-password">
        <Helmet title="Mijn Account | Wachtwoord wijzigen" />
        <Grid fluid>
          <Row>
            <Col xs={12}>
              <PageTitle>Wachtwoord wijzigen</PageTitle>
            </Col>
          </Row>
          <Row>
            <Col xs={12}>
              <p>
                Je kunt hier je wachtwoord voor je Tippiq Account wijzigen.
              </p>
              <br />
            </Col>
          </Row>
          <Row>
            <Col xs={12}>
              <ChangePasswordForm
                onSubmit={this.handleSubmit}
                errorMessage={password.error ? password.error.message : ''}
              />
            </Col>
          </Row>
          { passwordSaved &&
            <Row>
              <Col xs={12}>
                <span id="successMessage" className="text-primary">
                  Gelukt! Je wachtwoord is gewijzigd.
                </span>
              </Col>
            </Row>
          }
        </Grid>
      </div>
    );
  }
}

export default connect(
  state => ({
    password: state.password,
  }), dispatch => bindActionCreators({ changePassword }, dispatch)
)(ChangePassword);
