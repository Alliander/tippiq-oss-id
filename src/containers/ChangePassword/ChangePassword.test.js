import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ChangePassword } from './ChangePassword';

test('Basic render snapshot test', () => {
  const component = shallow(<ChangePassword
    error={'error'}
    changePassword={() => {}}
    location={{ query: '' }}
    password={{
      save: {
        success: '',
      },
      error: {
        message: '',
      },
    }}
  />);

  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

