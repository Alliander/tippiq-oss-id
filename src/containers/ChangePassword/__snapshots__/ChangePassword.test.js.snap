// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic render snapshot test 1`] = `
<div
  id="page-change-password"
>
  <HelmetWrapper
    title="Mijn Account | Wachtwoord wijzigen"
  />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <PageTitle>
          Wachtwoord wijzigen
        </PageTitle>
      </Col>
    </Row>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <p>
          Je kunt hier je wachtwoord voor je Tippiq Account wijzigen.
        </p>
        <br />
      </Col>
    </Row>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <ConnectedForm
          errorMessage=""
          onSubmit={[Function]}
        />
      </Col>
    </Row>
  </Grid>
</div>
`;
